homeassistant/components/samsungtv/media_player.py

```diff
@@ -147,15 +148,23 @@ def state(self):
         """Return the state of the device."""
         return self._state
 
+    @property
+    def available(self):
+        """Return the availability of the device."""
+        return self._state == STATE_ON
```

Comment:
Now the state can no longer be off, only either on or unavailable. That doesn't seem correct?

Reply:
Yes, your're right sounds weird, but when off we are not able to control the device. Except we have wol or something else configured.

Reply:
Underling library has WOL function implemented; dunno if it's the protocol used by "ha-samsungtv-smart" custom component, but this plugin is capable to switch on a TV that is off.

Reply:
We're not able to control the device when it's unavailable either.

Reply:
You're saying unavailable is the correct representation of the off state?

Reply:
@MartinHjelmare It's maybe not the correct representation but it's fact. My legacy device switches the network interfaces completely off, so I'm unable to wake it up. I've to use the remote or my self defined HDMI wake up signal from Kodi. To differentiate it a bit I could do it like this:
```python
    @property
    def available(self):
        """Return the availability of the device."""
        return self._state == STATE_ON or self._on_script
```

Reply:
Sounds good.